### PR TITLE
[PrismaCloudV2] - Add In and Out Mirroring 

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-5504](https://jira-hq.paloaltonetworks.local/browse/CIAC-5504).

## Description
Implementation of the mirror in and out for Prisma Cloud v2 integration, as described in this issue -  [CIAC-5504](https://jira-hq.paloaltonetworks.local/browse/CIAC-5504).

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
@yuvalbenshalom  - I'm still dealing with some pre-commit errors, and haven't managed to write unit-tests yet, but you can start the review.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 1]
```diff
@@ -595,6 +623,208 @@ def alert_to_incident_context(alert: Dict[str, Any]) -> Dict[str, Any]:
...
+    The second condition is mandatory so we will not send a re-opening request in case where the incident is updated so there is
+    a delta, but it is not the status that was changed.
+
+    Args:
+        delta: A dictionary of fields that changed from the last update - containing only the changed fields.
+
+    Returns: True - if re-opening the prisma alert is needed, False - otherwise.
+
+    """
+    return demisto.params().get('close_ticket') and 'closingUserId' in delta
```
Comment (yuvalbenshalom) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:789]: I would prefer to pass the specific part and not call it in internal command.
this is relevant to the other similar functions you introduced in the pr.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 2]
```diff
@@ -24,9 +24,41 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+    - Incoming And Outgoing - Changes in XSOAR incidents and in Prisma Cloud alerts will be reflected in both directions.
+    - None - Turns off incident mirroring.
+7. Optional: Check the *Close Mirrored XSOAR Incident* integration parameter to close or reopen the Cortex XSOAR incident when the corresponding alert is closed or re-opened in Prisma Cloud.
+8. Optional: Check the *Close Mirrored Prisma Cloud Alert* integration parameter to close or reopen the Prisma Cloud alert when the corresponding Cortex XSOAR incident is closed or re-opened.
+
+Newly fetched incidents will be mirrored in the chosen direction. However, this selection does not affect existing incidents.
+
+**Important Notes**
+
+- To ensure the mirroring works as expected, an incoming mapper is required, to map the expected fields in Cortex XSOAR.
```
Comment (yuvalbenshalom) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:56]: which mapper to pick?
maybe give an example as the default mapper.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good, see my comments.
there is a general comment about propagating parameters to inside functions instead of saving them in the client. lets talk about that.

File: Packs/PrismaCloud/IncidentFields/incidentfield-Prisma_Cloud_Status.json [hunk 3]
```diff
@@ -26,7 +26,7 @@
 	"group": 0,
 	"hidden": false,
 	"id": "incident_prismacloudstatus",
-	"isReadOnly": false,
+	"isReadOnly": true,
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/PrismaCloud/IncidentFields/incidentfield-Prisma_Cloud_Status.json:29]: Exactly. I want it to change only by the mirror In process. The field isn't mirrored out, so it makes no sense to let users edit it.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md (hunk 2 above)
Comment (ShacharKidor) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:56]: Added.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 4]
```diff
@@ -24,9 +24,42 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+
+Newly fetched incidents will be mirrored in the chosen direction. However, this selection does not affect existing incidents.
+
+**Important Notes**
+
+- To ensure the mirroring works as expected, an incoming mapper is required, to map the expected fields in Cortex XSOAR (you can use the default mapper - *Prisma Cloud - Incoming Mapper*).
+- When *mirroring in* incidents from Prisma Cloud to Cortex XSOAR:
+  - When enabling the *Close Mirrored XSOAR Incident* integration parameter, the field in Prisma Cloud that determines whether the incident was closed or re-opend is the `status` field.
+- When *mirroring out* incidents from Cortex XSOAR to Prisma Cloud:
+  - When enabling the *Close Mirrored Prisma Cloud Alert* integration parameter, the corresponding alert in Prisma Cloud will be closed with a *Dismissed* status for every reason chosen in the XSOAR incident (possible reasons are: `False Positive`, `Duplicate`, `Other` and `Resolved`). The *Reason* field of the Prisma Cloud alert will include the original reason selected in XSOAR and the close notes.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:60]:
```suggestion
  - When enabling the *Close Mirrored Prisma Cloud Alert* integration parameter, the corresponding alert in Prisma Cloud will be closed with a *Dismissed* status for every reason chosen in the Cortex XSOAR incident (possible reasons are: `False Positive`, `Duplicate`, `Other` and `Resolved`). The *Reason* field of the Prisma Cloud alert will include the original reason selected in Cortex XSOAR and the close notes.
```

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md [hunk 5]
```diff
@@ -24,9 +24,42 @@ If you are upgrading from a previous version of this integration, see [Breaking
...
+  - When re-opening an XSOAR incident with a `Resolved` Prisma Cloud status, the incident will be re-opened, but the alert in Prisma Cloud will remain Resolved due to API limitations.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/Integrations/PrismaCloudV2/README.md:61]:
```suggestion
  - When re-opening a Cortex XSOAR incident with a `Resolved` Prisma Cloud status, the incident will be re-opened, but the alert in Prisma Cloud will remain Resolved due to API limitations.  
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+
+#### Incident Fields
+
+##### Prisma Cloud Dismissal Note
+- Added the new **Prisma Cloud Dismissal Note** incident field.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:5]:
```suggestion
Added the new **Prisma Cloud Dismissal Note** incident field.
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+##### Prisma Cloud Reason
+- Updated the incident field to be read only (can be edited only by the mirror in functionality).
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:8]:
```suggestion
Updated the incident field to be read only (can be edited only by the mirror in functionality).
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+##### Prisma Cloud Status
+- Updated the incident field to be read only (can be edited only by the mirror in functionality).
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:11]:
```suggestion
Updated the incident field to be read only (can be edited only by the mirror in functionality).
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+- Added the following integration parameters:
+    - *Mirroring Direction*
+    - *Close Mirrored XSOAR Incident*
+    - *Close Mirrored Prisma Cloud Alert*
+
+#### Layouts
+
+##### AWS CloudTrail Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:29]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+#### Layouts
+
+##### AWS CloudTrail Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### AWS EC2 Instance Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:33]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### AWS EC2 Instance Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### AWS IAM Policy Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:37]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### AWS IAM Policy Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### GCP Compute Engine Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:41]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### GCP Compute Engine Misconfiguration Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### GCP Kubernetes Engine Misconfiguration
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:45]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### GCP Kubernetes Engine Misconfiguration
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### Prisma Cloud - VM Alert Prioritization
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:49]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### Prisma Cloud - VM Alert Prioritization
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+##### Prisma Cloud Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:53]:
```suggestion
Added the *Prisma Cloud Dismissal Note* field to the layout. 
```

File: Packs/PrismaCloud/ReleaseNotes/4_2_0.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+##### Prisma Cloud Incident
+
+- Added the ***Prisma Cloud Dismissal Note*** field to the layout. 
+
+#### Mappers
+
+##### Prisma Cloud - Incoming Mapper
+
+- Added the detection ***Prisma Cloud Dismissal Note*** field, and the mirroring fields to the Mapper.
```
Comment (ShirleyDenkberg) [inline Packs/PrismaCloud/ReleaseNotes/4_2_0.md:59]:
```suggestion
Added the detection *Prisma Cloud Dismissal Note* field, and the mirroring fields to the Mapper.
```

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom Doc review completed.

File: Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py [hunk 17]
```diff
@@ -595,6 +623,208 @@ def alert_to_incident_context(alert: Dict[str, Any]) -> Dict[str, Any]:
...
+def set_xsoar_incident_entries(updated_object: Dict[str, Any], remote_alert_id: str) -> Dict | None:
+    """
+    Extracts the status of the mirrored Prisma alert, and close/reopen the matched XSOAR incident in accordance.
+    Args:
+        updated_object: A dictionary contains the mirrored relevant fields.
+        remote_alert_id: The id of the mirrored Prisma Cloud alert.
+
+    Returns: An entry object with relevant data for closing or reopening an XSOAR incident.
+    """
+    if demisto.params().get('close_incident'):
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/PrismaCloud/Integrations/PrismaCloudV2/PrismaCloudV2.py:717]: Agree - changed that.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**